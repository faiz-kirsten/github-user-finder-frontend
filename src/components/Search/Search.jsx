import style from "./Search.module.css"; // import css

// search input form that triggers the handleForm function on submit
const Search = ({ handleForm, setUsername }) => {
    return (
        <form onSubmit={handleForm} className={style.form}>
            <div className={style.formController}>
                <input
                    placeholder="Enter username..."
                    onChange={(e) => {
                        setUsername(e.target.value);
                    }}
                    className={style.searchInput}
                    required
                />

                <input type="submit" className="btn" />
            </div>
        </form>
    );
};

export default Search;
