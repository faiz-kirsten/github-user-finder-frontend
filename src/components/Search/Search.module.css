/* form styling */

.form {
    display: grid;
    place-items: center;
    margin-bottom: 2em;
    width: 100%;
}

.formController {
    display: flex;
    /* gap: 2em; */
    background-color: #333;
    justify-content: space-around;
    border: 1px solid #333;
    padding: 0.4em 0.5em;
    border-radius: 2em;
}

.searchInput {
    border: none;
    width: 25em;
    font-size: 1.2rem;
    margin-left: 0.8em;
    background-color: #333;
    color: #fff;
}

.searchInput:focus {
    outline: none;
}
