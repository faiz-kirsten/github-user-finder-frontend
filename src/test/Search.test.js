import React from "react";
import Search from "../components/Search/Search";
import renderer from "react-test-renderer";

const mockHandleForm = jest.fn();
const mockSetUsername = jest.fn();

test("Renders input component correctly with expected props", () => {
    const tree = renderer
        .create(
            <Search handleForm={mockHandleForm} setUsername={mockSetUsername} />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
