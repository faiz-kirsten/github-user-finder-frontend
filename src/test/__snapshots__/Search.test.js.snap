// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders input component correctly with expected props 1`] = `
<form
  className="form"
  onSubmit={[MockFunction]}
>
  <div
    className="formController"
  >
    <input
      className="searchInput"
      onChange={[Function]}
      placeholder="Enter username..."
      required={true}
    />
    <input
      className="btn"
      type="submit"
    />
  </div>
</form>
`;
